app/models/course.rb

```diff
@@ -172,6 +174,45 @@ def allocations_report_fields
     ]
   end
 
+  def spreadsheet_columns
+    [
+      ['Academic Year', '2020/21'],
+      ['Requested By (Name)', accrediting_provider&.provider_name],
+      ['Requested by (UKPRN)', ''],
+      ['Partner ITT Provider (Name)', provider.provider_name],
+      ['Partner ITT provider (UKPRN)', provider.organisations.first&.school_nctl_organisation&.urn],
+      ['Allocation Subject', allocation_subjects.join(' | ')],
+      ['Route', program_type],
+      ['Course Aim', qualification],
+      ['Course Level', 'PG'],
+      ['Min. no. of recruits', ''],
+      ['Forecast no. of recruits ', ''],
+      ['3 Year Intention', ''],
+      ['Awarding Institution', ''],
+      ['Other Institution', '']
+    ]
+  end
+
+  # Outputs an allocations XLSX with a randomly generated suffix to `public/`.
+  # The first argument is an optional prefix for the filename to make it
+  # easier to differentiate among ~200 other similar XLSX files. Defaults to
+  # the provider.provider_code.
+  # The filename will have a UUID prefix so that it can be hosted
+  # and linked to from an Azure bucket, but that the names can't be
+  # simply guessed.
+  # Example usage:
+  #    $ bin/rails c
+  #    > Course.save_xlsx(Course.first(20))
+  def self.save_allocations_report(file_name_prefix = all.first.provider.provider_code)
+    file_data = all.to_xlsx
+
+    file_name_suffix = SecureRandom.uuid
```

Comment:
timestamp instead?

Reply:
Timestamp could be ~the same for an export of 200 or so spreadsheets, reducing its effectiveness in making the URL harder to guess 🤔